tests/common/dualtor/dual_tor_utils.py

```diff
@@ -0,0 +1,90 @@
+import logging
+import pytest
+
+logger = logging.getLogger(__name__)
+
+@pytest.fixture(scope='session')
+def tor_mux_intf(duthosts):
+    '''
+    Returns the server-facing interface on the ToR to be used for testing 
+    '''
+    # The same ports on 
+    dut = duthosts[0]
+    return dut.get_vlan_intfs()[0]
+
+
+@pytest.fixture(scope='session')
+def ptf_server_intf(duthosts, tor_mux_intf, tbinfo):
+    '''
+    Returns the ToR-facing interface on the PTF to be used for testing
+
+    This should be connected to the interface returned by `tor_mux_intf`
+    '''
+    mg_facts = duthosts[0].get_extended_minigraph_facts(tbinfo)
+    ptf_port_index = mg_facts['minigraph_ptf_indices'][tor_mux_intf]
+    ptf_intf = 'eth{}'.format(ptf_port_index)
+
+    logger.info("Using PTF server interface {} for test".format(ptf_intf))
+    return ptf_intf
+
+
+@pytest.fixture(scope='session')
+def t1_upper_tor_intfs(upper_tor_host, tbinfo):
```

Comment:
For testing purposes, my understanding is upper/lower ToR correspond to active/standby ToR. If that is correct, our active and standby ToRs change with different scenarios playing.
In this case having a `session` level fixture will always return the same value throughout the test. This may cause an issue as within module scope, the upper can become lower and vice-versa.
Applies to every upper/lower utility here.

Reply:
I believe the upper/lower designations are fixed and do not change throughout the tests, they are simply used to distinguish the two ToRs used.

Reply:
Even I believed so, but then I came across this HLD saying `For testing purposes, it's assumed that the upper ToR is the active ToR at the beginning of the test`. That made me think upper translates to active:
https://github.com/theasianpianist/sonic-mgmt/blob/dual-tor-test-plan/docs/testplan/dual_tor/dual_tor_test_plan_action_items.md

Reply:
That's a good catch, I was not clear enough when writing the doc. The upper/lower designation is actually set by the minigraph (the minigraph generator determines upper/lower by hostname which is the same method used here), and should not change unless a new minigraph is deployed. I have edited the doc for clarity.